### PR TITLE
Add syntactic classification for doc comments.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/services/services.ts [hunk 1]
```diff
@@ -2932,13 +2936,29 @@ module ts {
...
                         }
-                    });
+                    }
+
+                    if (isJavaScriptFile && type.flags & TypeFlags.Union) {
+                        // In javascript files, for union types, we don't just get the members that 
+                        // the individual types have in common, we also include all the members that
+                        // each individual type has.  This is because we're going to add all identifiers
+                        // anyways.  So we might as well elevate the members that were at least part
+                        // of the individual types to a higher status than since we know what they are.
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: -than

File: src/services/services.ts [hunk 2]
```diff
@@ -6118,6 +6138,90 @@ module ts {
                 }
             }
 
+            function classifyComment(token: Node, kind: SyntaxKind, start: number, width: number) {
+                if (kind === SyntaxKind.MultiLineCommentTrivia) {
+                    // See if this is a doc comment.  If so, we'll classify certain portions of it
+                    // specially.
+                    let jsDocComment = parseIsolatedJSDocComment(sourceFile.text, start, width);
+                    if (jsDocComment && jsDocComment.jsDocComment) {
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: What is even going on here? What is `jsDocComment.jsDocComment`? When will it not be set?

File: src/services/services.ts [hunk 3]
```diff
@@ -6118,6 +6138,90 @@ module ts {
...
+
+            function pushCommentRange(start: number, width: number) {
+                pushClassification(start, width, ClassificationType.comment);
+            }
+
+            function classifyJSDocComment(docComment: JSDocComment) {
+                let pos = docComment.pos;
+
+                for (let tag of docComment.tags) {
+                    if (tag.pos !== pos) {
```
Comment (DanielRosenwasser) [inline src/services/services.ts]: What is this check for?

**Comment by DanielRosenwasser** [comment]
I think we could use some more coverage.

You classify types as in a regular context again. What happens in the following cases?

```TypeScript
/**
 * @param foo { (x: number) => string }
 */
function f(foo) {
}
```

```TypeScript
// note the '/*'
/**
 * @param foo { Foo /* } 
 * @param bar
 */
function f(foo, bar) {
}
```

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**